libwayshot/src/lib.rs

```diff
@@ -446,13 +447,50 @@ impl WayshotConnection {
             }
         };
 
+        let shm = self.globals.bind::<WlShm, _, _>(&qh, 1..=1, ())?;
         for (frame_copy, frame_guard, output_info) in frames {
             tracing::span!(
                 tracing::Level::DEBUG,
                 "overlay_frames::surface",
                 output = format!("{output_info}")
             )
             .in_scope(|| -> Result<()> {
+                let buffer = {
+                    let output_size = output_info.logical_region.inner.size;
+                    if output_size != frame_copy.frame_format.size {
+                        let file = tempfile::tempfile().unwrap();
+                        let image: DynamicImage = frame_copy.try_into()?;
+                        let image = image::imageops::resize(
+                            &image,
+                            output_size.width,
+                            output_size.height,
+                            image::imageops::FilterType::Triangle,
+                        );
+                        init_overlay(&image, &file);
+                        let pool = shm.create_pool(
+                            file.as_fd(),
+                            frame_copy
+                                .frame_format
+                                .byte_size()
+                                .try_into()
+                                .map_err(|_| Error::BufferTooSmall)?,
+                            &qh,
+                            (),
+                        );
+
+                        pool.create_buffer(
+                            0,
+                            output_size.width as i32,
+                            output_size.height as i32,
+                            (output_size.width * 4) as i32,
+                            frame_copy.frame_format.format,
+                            &qh,
+                            (),
+                        )
```

Comment:
There should be no reason to resize. We have a pixel perfect buffer already that represents every single pixel shown on the screen.

Why do you think you need to do this?

Reply:
when the scale is int, I will not resize it, but when it is fractal, I wll resize it manully